emu/src/arm/arm7tdmi.rs

```diff
@@ -43,15 +44,28 @@ impl Registers {
     }
 }
 
-#[derive(Default)]
 pub struct Arm7tdmi {
     pub(crate) rom: Arc<Mutex<Vec<u8>>>,
     pub(crate) memory: Arc<Mutex<InternalMemory>>,
 
-    pub cpsr: Cpsr,
+    pub cpsr: Psr,
     pub registers: Registers,
 }
 
+impl Default for Arm7tdmi {
+    fn default() -> Self {
+        // SPSR are initalized with the value related to their respective
+        // mode (5 least significant bits).
+        // e.g., FIQ SPSR has mode 0b10001, IRQ 0b10010, etc...
+        Self {
+            rom: Arc::new(Mutex::new(Vec::default())),
+            memory: Arc::new(Mutex::new(InternalMemory::default())),
+            cpsr: Psr::from(Mode::User), // FIXME: Starting as User? Not sure
```

Comment:
cpsr -> psr

Reply:
I think `cpsr` is okay since this variable in particular is the current psr